src/Compilers/CSharp/Test/Semantic/Semantics/RefExtensionMethodsTests.cs

```diff
@@ -38,6 +38,136 @@ public static void Main()
                 Diagnostic(ErrorCode.ERR_RefLvalueExpected, "5").WithLocation(13, 9));
         }
 
+        [Fact]
+        public void ExtensionMethods_StructCollectionInitializer()
+        {
+            var code = @"
+public struct MyStruct : System.Collections.IEnumerable
+{
+    public int i;
+    public System.Collections.IEnumerator GetEnumerator() => throw new System.NotImplementedException();
+}
+
+public static class MyStructExtension
+{
+    public static void Add(ref this MyStruct s, int i)
+    {
+        s.i += i;
+    }
+}
+
+public static class Program
+{
+    public static void Main()
+    {
+        var s = new MyStruct { 1 };
+        System.Console.Write(s.i);
+    }
+}";
+            CompileAndVerify(code, expectedOutput: "1");
+        }
+
+        [Fact]
+        public void ExtensionMethods_StructCollectionInitializerInParam()
+        {
+            var code = @"
+public struct MyStruct : System.Collections.IEnumerable
+{
+    public int i;
+    public System.Collections.IEnumerator GetEnumerator() => throw new System.NotImplementedException();
+}
+
+public static class MyStructExtension
+{
+    public static void Add(ref this MyStruct s, in MyStruct other)
+    {
+        s.i += other.i;
+    }
+}
+
+public static class Program
+{
+    public static void Main()
+    {
+        var other = new MyStruct { i = 2 };
+        var s = new MyStruct { other };
+        System.Console.Write(s.i);
+    }
+}";
+            var verifier = CompileAndVerify(code, expectedOutput: "2");
+            verifier.VerifyIL("Program.Main", @"{
+  // Code size       47 (0x2f)
+  .maxstack  2
+  .locals init (MyStruct V_0, //other
+                MyStruct V_1)
```

Comment:
In the future consider using different types here. It makes reading the IL a bit easier. Change is fine though as is.